File: crypto_agent/notify.py
import aiohttp


class Notifier:
    def __init__(self, telegram_token: str = "", telegram_chat_id: str = ""):
        self.telegram_token = telegram_token
        self.telegram_chat_id = telegram_chat_id

    async def send(self, message: str):
        if self.telegram_token and self.telegram_chat_id:
            await self._send_telegram(message)

    async def _send_telegram(self, message: str):
        url = f"https://api.telegram.org/bot{self.telegram_token}/sendMessage"
        try:
            async with aiohttp.ClientSession() as session:
                await session.post(url, json={
                    "chat_id": self.telegram_chat_id,
                    "text": message[:4000],
                    "parse_mode": "Markdown",
                })
        except Exception:
            pass

    @property
    def enabled(self) -> bool:
        return bool(self.telegram_token and self.telegram_chat_id)

File: src/notify.ts
export class Notifier {
  private telegramToken: string;
  private telegramChatId: string;

  constructor(telegramToken = "", telegramChatId = "") {
    this.telegramToken = telegramToken;
    this.telegramChatId = telegramChatId;
  }

  async send(message: string): Promise<void> {
    if (this.telegramToken && this.telegramChatId) {
      await this.sendTelegram(message);
    }
  }

  private async sendTelegram(message: string): Promise<void> {
    const url = `https://api.telegram.org/bot${this.telegramToken}/sendMessage`;
    try {
      await fetch(url, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({
          chat_id: this.telegramChatId,
          text: message.slice(0, 4000),
          parse_mode: "Markdown",
        }),
      });
    } catch {
      // silently ignore
    }
  }

  get enabled(): boolean {
    return Boolean(this.telegramToken && this.telegramChatId);
  }
}
